src/PhpSpreadsheet/Writer/Xls.php

```diff
@@ -389,25 +389,129 @@ private function buildWorksheetEschers(): void
+        $blip = new Blip();
+        $blip->setData($blipData);
+
+        $BSE = new BSE();
+        $BSE->setBlipType($blipType);
+        $BSE->setBlip($blip);
+
+        $bstoreContainer->addBSE($BSE);
+    }
+
+    private function processDrawing(BstoreContainer &$bstoreContainer, BaseDrawing $drawing): void
+    {
+        $blipData = '';
+        $filename = $drawing->getPath();
+
+        [$imagesx, $imagesy, $imageFormat] = getimagesize($filename);
+
+        switch ($imageFormat) {
+            case 1: // GIF, not supported by BIFF8, we convert to PNG
+                $blipType = BSE::BLIPTYPE_PNG;
+                ob_start();
+                imagepng(imagecreatefromgif($filename));
+                $blipData = ob_get_contents();
+                ob_end_clean();
+
+                break;
+            case 2: // JPEG
+                $blipType = BSE::BLIPTYPE_JPEG;
+                $blipData = file_get_contents($filename);
+
+                break;
+            case 3: // PNG
+                $blipType = BSE::BLIPTYPE_PNG;
+                $blipData = file_get_contents($filename);
+
+                break;
+            case 6: // Windows DIB (BMP), we convert to PNG
+                $blipType = BSE::BLIPTYPE_PNG;
+                ob_start();
+                imagepng(SharedDrawing::imagecreatefrombmp($filename));
+                $blipData = ob_get_contents();
+                ob_end_clean();
+
+                break;
+        }
+        if ($blipData) {
+            $blip = new Blip();
+            $blip->setData($blipData);
+
+            $BSE = new BSE();
+            $BSE->setBlipType($blipType);
+            $BSE->setBlip($blip);
+
+            $bstoreContainer->addBSE($BSE);
+        }
+    }
+
+    private function processBaseDrawing(BstoreContainer &$bstoreContainer, BaseDrawing $drawing): void
+    {
+        if ($drawing instanceof Drawing) {
+            $this->processDrawing($bstoreContainer, $drawing);
+        } elseif ($drawing instanceof MemoryDrawing) {
+            $this->processMemoryDrawing($bstoreContainer, $drawing, $drawing->getRenderingFunction());
+        }
+    }
+
     /**
-     * Build the Escher object corresponding to the MSODRAWINGGROUP record.
+     * Total number of shared strings in workbook.
+     *
+     * @var bool
      */
-    private function buildWorkbookEscher(): void
+    private $simulateNoGd = false;
+
+    public function setSimulateNoGd(bool $arg): void
     {
-        $escher = null;
+        $this->simulateNoGd = $arg;
+    }
 
+    private function checkForDrawings(): bool
+    {
         // any drawings in this workbook?
         $found = false;
         foreach ($this->spreadsheet->getAllSheets() as $sheet) {
             if (count($sheet->getDrawingCollection()) > 0) {
                 $found = true;
+                if ($this->simulateNoGd || !extension_loaded('gd')) {
+                    throw new RuntimeException('Saving images in xls requires gd extension');
+                }
```

Comment:
This, and `setSimulateNoGd` and related, should be entirely deleted, because we now use composer to assert for us that all dependencies do exist. And thus our code can assume it is there.

```suggestion
```

Reply:
That does seem like a much better way to handle it. I have included your suggested change in my latest commit/push.